src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -1169,6 +1169,22 @@ def _get_adjacent_exon(
         :param end: Genomic coordinate of breakpoint
         :return: Exon number corresponding to adjacent exon. Will be 0-based
         """
+        # Check if a breakpoint occurs before/after the transcript boundaries
+        bp = start if start else end
+        if strand == Strand.POSITIVE:
+            if bp < tx_exons_genomic_coords[0].alt_start_i:
+                return 0
+            if bp > tx_exons_genomic_coords[len(tx_exons_genomic_coords) - 1].alt_end_i:
```

Comment:
Since we are calling `len(tx_exons_genomic_coords) - 1` 5 times in this method, consider storing this value as a variable